x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1391,7 +1412,14 @@ private Expression combine(Or or) {
                 }
             }
 
-            return changed ? Predicates.combineOr(CollectionUtils.combine(exps, bcs, ranges)) : or;
+            Boolean updated = filterDisjunctionByNotEquals(notEquals, bcs, ranges);
```

Comment:
I would prefer the return value of the function to be `boolean` and not `Boolean` since false is never returned currently. Or if I'm confused if false is also a valid return value then why do we return a below literal with `Boolean.TRUE` instead of using the `updated` variable there?

Reply:
`false` will be returned in case a `NotEquals` and a `BinaryComparison` aren't semantically equal (and can't be simplified otherwise either; smth like `a != 2 OR b < 3`).
So I'd need a three states-return: one for no optimisation possible, one for optimisation has been applied and another one when the entire expression evaluates to literal `TRUE`.
I thought an enum might be too much, but happy to apply suggestions.

Reply:
Ok, thank you, I missed that somehow.